File: vscode/test/input-display-mode.test.js
const assert = require('node:assert/strict');
const test = require('node:test');

const { InputDisplayMode, inputDisplayModeFor } = require('../out/core/types');

test('display mode depends only on input state', () => {
  assert.equal(inputDisplayModeFor({ isAsciiMode: true, isCapsLock: false }), InputDisplayMode.ENGLISH);
  assert.equal(inputDisplayModeFor({ isAsciiMode: false, isCapsLock: false }), InputDisplayMode.CHINESE);
  assert.equal(inputDisplayModeFor({ isAsciiMode: true, isCapsLock: true }), InputDisplayMode.CAPS);
  assert.equal(inputDisplayModeFor({ isAsciiMode: false, isCapsLock: true }), InputDisplayMode.CAPS);
});

File: core/src/test/kotlin/com/auto_switch_ime/core/InputDisplayModeTest.kt
package com.auto_switch_ime.core

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test

class InputDisplayModeTest {

    @Test
    fun `display mode depends only on input state`() {
        assertEquals(InputDisplayMode.ENGLISH, ImeState(true, false).toInputDisplayMode())
        assertEquals(InputDisplayMode.CHINESE, ImeState(false, false).toInputDisplayMode())
        assertEquals(InputDisplayMode.CAPS, ImeState(true, true).toInputDisplayMode())
        assertEquals(InputDisplayMode.CAPS, ImeState(false, true).toInputDisplayMode())
    }
}
